EventListener/SoftDeleteListener.php

```diff
@@ -137,15 +148,11 @@ public function preSoftDelete(LifecycleEventArgs $args)
                     }
 
                     if ($objects) {
-                        $factory = $em->getMetadataFactory();
-                        $cacheDriver = $factory->getCacheDriver();
-                        $cacheId = ExtensionMetadataFactory::getCacheId($namespace, 'Gedmo\SoftDeleteable');
-                        $softDelete = false;
-                        if (($config = $cacheDriver->fetch($cacheId)) !== false) {
-                            $softDelete = isset($config['softDeleteable']) && $config['softDeleteable'];
-                        }
+                        $reflectionClass = new \ReflectionClass($namespace);
+                        $classAnnotation = $this->reader->getClassAnnotation($reflectionClass, \Gedmo\Mapping\Annotation\SoftDeleteable::class);
+                        $softDelete = $classAnnotation !== null && $classAnnotation->hardDelete !== true;
```

Comment:
Should more likely look like: 
`$softDelete = $classAnnotation instanceof \Gedmo\Mapping\Annotation\SoftDeleteable;`?

`hardDelete` is another option. See https://github.com/doctrine-extensions/DoctrineExtensions/blob/main/doc/softdeleteable.md (Section: SoftDeleteable attributes)

Reply:
Hello, @felixrupp . Thank you for your comment!

I prefer avoiding `instanceof` if possible. In what way would that be better than this proposal? Are we missing something here? I have assumed that the ` Doctrine\Common\Annotations\Reader` is the correct class to ask for information about annotations.

P.S.: Sorry for replying late, there were vacations and quite busy times in general. We wanted to prepare an "official" MR once we are ready, but you seem to have noticed our fork before we did so.

Reply:
@felixrupp Hello, I have read your message again. I still thing that the implementation here is correct – the `\Doctrine\Common\Annotations\Reader::getClassAnnotation` either returns the requested annotation, or null (as said in it's documentation).

Therefore `$classAnnotation instanceof \Gedmo\Mapping\Annotation\SoftDeleteable` should be exactly equal to `$classAnnotation !== null` in this case. 

And the `&& $classAnnotation->hardDelete !== true;` is required, because the parameter `$softDelete` is only used in the `SoftDeleteListener::processOnDeleteCascadeOperation` method. And there you can see that it serves exactly this purpose - "do a soft, or hard delete?".

_It took us a while to get back to this, sorry._